skxray/img_proc/mathops.py

```diff
@@ -0,0 +1,349 @@
+
+        In the event that bitwise operations are required the operators &,
+        |, ^, ~ may also be used, though I'm struggling to come up with a
+        scenario where this will be used.
+
+        Order of operations and parenthesis are taken into account when
+        evaluating the expression.
+
+    A : {ndarray, int, float}
+        Data set or constant to be offset or manipulated
+
+    B : {ndarray, int, float}
+        Data set or constant to be offset or manipulated
+
+    C : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    D : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    E : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    F : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    G : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    H : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+
+    Returns
+    -------
+    output : {ndarray, int, float}
+        Returns the resulting array or value to the designated variable
+
+    Example
+    -------
+    result = mathops.arithmetic_custom('(A+C)/(B+D)', img_1, img_2, 2, 4)
+    """
+    output = eval(parser.expr(expression).compile())
+    return output
+
+
+def logic_basic(operation,
+                src_data1,
+                src_data2=None):
+    """
+    This function enables the computation of the basic logical operations
+    oft used in image processing of two image or volume  data sets. This
+    function can be used for data comparison, material isolation,
+    noise removal, or mask application/generation.
+
+    Parameters
+    ----------
+    operation : str
+        options include:
+            'and' -- 2 inputs
+            'or' -- 2 inputs
+            'not' -- 1 input
+            'xor' -- 2 inputs
+            'nand' -- 2 inputs
+            'subtract' -- 2 inputs
+
+    src_data1 : {ndarray, int, float, list, tuple}
+        Specifies the first reference
+
+    src_data2 : {ndarray, int, float, list, tuple}
+        Specifies the second reference
+
+    Returns
+    -------
+    output : {ndarray, bool}
+        Returns the result of the logical operation, which can be an array,
+        or a simple boolean result.
+
+    Example
+    -------
+    result = mathops.logic_basic('and', img_1, img_2)
+    """
+    logic_dict = {'and' : np.logical_and,
+                  'or' : np.logical_or,
+                  'not' : np.logical_not,
+                  'xor' : np.logical_xor,
+                  'nand' : logical_NAND,
+                  'nor' : logical_NOR,
+                  'subtract' : logical_SUB
+                  }
+    output = logic_dict[operation](src_data1,
+                                   src_data2)
+    return output
+
+
+def logical_NAND(src_data1,

```

Comment:
I would suggest that you keep the pattern consistent from the numpy by making `NAND`, `NOR` and `SUB` lowercase for these three functions